File: src/main/java/com/example/chatService/entity/Participant.java
package com.example.chatService.entity;

import com.example.chatService.dto.RoomRole;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import lombok.Getter;
import lombok.Setter;

import java.time.OffsetDateTime;

@Entity
@Getter @Setter
public class Participant {

    @Id
    @GeneratedValue
    private Long id;
    private String roomId;
    private String userId;
    private RoomRole role;
    private boolean isActive;

    private OffsetDateTime joinedAt;
    private OffsetDateTime leftAt;

    public static Participant join(String roomId, String userId) {
        Participant p = new Participant();
        p.roomId = roomId;
        p.userId = userId;
        p.isActive = true;
        p.joinedAt = OffsetDateTime.now();
        return p;
    }

    public void leave() {
        this.isActive = false;
        this.leftAt = OffsetDateTime.now();
    }
}

File: src/main/java/com/example/chatService/entity/Participant.kt
package com.example.chatService.entity

import com.example.chatService.dto.RoomRole
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.Id
import java.time.OffsetDateTime

@Entity
class Participant(
    @Id
    @GeneratedValue
    var id: Long? = null,
    var roomId: String? = null,
    var userId: String = "",
    var role: RoomRole? = null,
    var isActive: Boolean = false,
    var joinedAt: OffsetDateTime? = null,
    var leftAt: OffsetDateTime? = null
) {
    fun leave() {
        isActive = false
        leftAt = OffsetDateTime.now()
    }

    companion object {
        fun join(roomId: String, userId: String): Participant {
            return Participant().apply {
                this.roomId = roomId
                this.userId = userId
                this.isActive = true
                this.joinedAt = OffsetDateTime.now()
            }
        }
    }
}
